### PR TITLE
Prepare for release v0.5.1

### PR DESCRIPTION
**Pull request by bitzoic**
## Type of change

- Prepare for New Release

## Changes

The following changes have been made:

- Updates CHANGELOG.md with the changes for this release
- Updates README with new version
- Updates docs with new version

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.
- [x] I have updated the changelog to reflect the changes on this PR.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,25 +5,25 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
-## [Unreleased] - yyyy-mm-dd
+## [Version 0.52.0] - 2024-06-12
```
Comment (K1-R1) [inline CHANGELOG.md:8]:
```suggestion
## [Version 0.5.1] - 2024-06-12
```

**Comment by bitzoic** [comment]
This PR should wait on https://github.com/FuelLabs/sway-standards/pull/110

**Review by K1-R1 — APPROVED** [review]

**Review by K1-R1 — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,24 +5,25 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
-## [Unreleased] - yyyy-mm-dd
+## [Version 0.52.1] - 2024-06-13
```
Comment (SwayStar123) [inline CHANGELOG.md:8]: 52? not 5?

Comment (SwayStar123) [inline CHANGELOG.md:8]:
```suggestion
## [Version 0.5.1] - 2024-06-13
```

**Review by SwayStar123 — APPROVED** [review]

**Review by SwayStar123 — DISMISSED** [review]

**Comment by K1-R1** [comment]
Can the changelog for #112 be added as it was missing from the PR itself

**Review by SwayStar123 — APPROVED** [review]

**Review by K1-R1 — APPROVED** [review]

**Pull request merged**